contracts/terraswap_factory/src/contract.rs

```diff
@@ -236,21 +255,74 @@ pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> StdResult<Response> {
     let pair_contract = res.get_address();
     let pair_info = query_pair_info_from_pair(&deps.querier, Addr::unchecked(pair_contract))?;
 
+    let raw_infos = [
+        tmp_pair_info.assets[0].info.clone(),
+        tmp_pair_info.assets[1].info.clone(),
+    ];
+
     PAIRS.save(
         deps.storage,
         &tmp_pair_info.pair_key,
         &PairInfoRaw {
             liquidity_token: deps.api.addr_canonicalize(&pair_info.liquidity_token)?,
             contract_addr: deps.api.addr_canonicalize(pair_contract)?,
-            asset_infos: tmp_pair_info.asset_infos,
+            asset_infos: raw_infos,
             asset_decimals: tmp_pair_info.asset_decimals,
         },
     )?;
 
-    Ok(Response::new().add_attributes(vec![
-        ("pair_contract_addr", pair_contract),
-        ("liquidity_token_addr", pair_info.liquidity_token.as_str()),
-    ]))
+    let mut messages: Vec<CosmosMsg> = vec![];
+    if !tmp_pair_info.assets[0].amount.is_zero() || !tmp_pair_info.assets[1].amount.is_zero() {
```

Comment:
Q: Is it also possible to not provide a token on one side of the pair?

Reply:
Since the share of provide-liquidity will be 0, it will be reverted when MINIMUM_LIQUIDITY is deducted.